### ISSUE TITLE
Problem with text/event-stream when compress=true

### ISSUE THREAD

**Issue by opavlov24** [containous/traefik#2576]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

We use [concourse](https://concourse.ci/) (it's a CI/CD tool) under traefik. Concourse UI uses `text/stream` to get information about builds from its backend. In traefik 1.3.8 everything works fine, but after updating to traefik 1.4.0 - 1.4.5 concourse can't get any event from its backend (infinite waiting...).

### What did you expect to see?

Concourse gets events from the backend.

### What did you see instead?

Infinite waiting.

### Output of `traefik version`: (_What version of Traefik are you using?_)

Traefik 1.4.0 - 1.4.5

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

The route to concourse was configured statically.
```toml
################################################################
# Global configuration
################################################################

logLevel = "DEBUG"

# Traefik logs file
traefikLogsFile = "/var/log/traefik-debug.log"

# Access logs file
accessLogsFile = "/var/log/accessTraefik.log"

# Entrypoints
defaultEntryPoints = ["http", "https"]
[entryPoints]
  [entryPoints.http]
  address = ":80"
	[entryPoints.http.redirect]
	entryPoint = "https"
  [entryPoints.https]
  address = ":443"
  compress = true
    [entryPoints.https.tls]
      [[entryPoints.https.tls.certificates]]
      certFile = ""
      keyFile = ""

# ...

[file]

[backends]
  [backends.concourse]
    [backends.concourse.servers.server1]
    url = "http://some-ip:8080"
# ...

[frontends]
  [frontends.concourse]
  backend = "concourse"
  passHostHeader = true
    [frontends.concourse.routes.host]
    rule = "Host:some-host"
```

### If applicable, please paste the log output in debug mode (`--debug` switch)

Only these lines are related to concourse in traefik logs.

```shell
time="2017-12-15T22:13:12Z" level=debug msg="Round trip: http://some-ip:port, code: 200, duration: 6.000457ms tls:version: 303, tls:resume:false, tls:csuite:c02f, tls:server:some-host"
time="2017-12-15T22:13:14Z" level=debug msg="Round trip: http://some-ip:port, code: 200, duration: 2.51749ms tls:version: 303, tls:resume:false, tls:csuite:c02f, tls:server:some-host"
time="2017-12-15T22:13:14Z" level=debug msg="Round trip: http://some-ip:port, code: 200, duration: 3.597858ms tls:version: 303, tls:resume:false, tls:csuite:c02f, tls:server:some-host"
```

**Comment by ldez** [comment]
Could try with 1.5.0-rc2 ?

**Comment by opavlov24** [comment]
I tried. The same thing.

**Comment by Juliens** [comment]
Hi @opavlov24 , can you give me more information about your use case because I have tried to install a concourse stack, but I'm not sure where is the real issue (where in the web UI).

Can you try with the last RC release too (1.5.0-rc4) ?

**Comment by TheCase** [comment]
After you log in to the Concourse UI, click on a pipeline event (one that has logs).  The log will not display, instead it gives you a link to the login page.

Check your browser console.  You'll see an error about the text/stream event.

**Comment by TheCase** [comment]
To follow up, I tested with 1.5.0-rc4 and there is no longer an issue related to displaying the logs for pipeline events in Concourse.

**Comment by Juliens** [comment]
Good to know.
Thank you

**Comment by mandrean** [comment]
Hi @TheCase and @Juliens. We run traefik `v1.5.4` and concourse `v3.9.0` but still have this issue.

Any thoughts?

**Comment by Juliens** [comment]
Hi @mandrean, 
Could you open a new issue with your configuration, and everything needed in order to reproduce your problem?
Thx

**Comment by opavlov24** [comment]
Hi everyone!
I have new information about the problem. So, we discovered that traefik1.5.4 works with SSE (we write a test-service with SSE support), but it hasn't worked with Concourse 3.9.2.

**Comment by gbgdev** [comment]
Hey @TheCase, what version of Concourse you were using when you saw the events displayed successfully on `1.5.0-rc4`?

**Comment by TheCase** [comment]
my working Concourse version is 3.8.0

**Comment by emcniece** [comment]
Reporting in with Traefik 1.5.4-1 and Concourse 3.11.0. Not only are build logs not loading, the entire web UI is timing out. Currently testing to determine if this is a red herring.

**Comment by emcniece** [comment]
Ok so the web container timeouts were due to a taxed instance, but I am reliably experiencing the lack of build logs. Confirmed that Traefik is the cause - this works fine with an Nginx proxy.

Devtools network panel shows that the `events` API call is the one that hangs with Traefik. With Nginx this call completes with a 200 and returns content, but with Traefik it hangs and never completes. I have tested and confirmed this via `cURL` by copying the request in curl form from devtools and executing it in terminal:

```
curl 'https://ci.site.com/api/v1/builds/1352/events' -H 'Accept: text/event-stream' -H 'Cache-Control: no-cache' -H 'cookie: __utmz=xx.xx.xx.xx.utmcsr=linkedin.com|utmccn=(referral)|utmcmd=referral|utmcct=/feed/; hubspotutk=xx; __utmc=xx; __hssrc=1; _ga=GA1.2.xx.xx; __utma=xx.xx.xx.xx.xx.xx; __hstc=xx.xx.xx.xx.xx.xx; ATC-Authorization="Bearer xxxxxx"' -H 'Referer: https://ci.site.com/teams/main/pipelines/mobile-test/jobs/test-pr/builds/7' -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/65.0.3325.181 Safari/537.36' --compressed
```

A quick search reveals https://github.com/containous/traefik/issues/819 which mentioned that perhaps the `Content-Type` header requires `charset=uft-8` at the end, but this didn't change anything for me. I also tried adding `-H "transfer-encoding:chunked"` from https://github.com/containous/traefik/issues/503 to no avail.

**Comment by Juliens** [comment]
I tried with concourse 3.9.2.
I follow https://concoursetutorial.com/ and https://concoursetutorial.com/basics/task-hello-world/ to install my concourse and create a build.

Then I launched a Traefik 1.5.4 (https://github.com/containous/traefik/releases/download/v1.5.4/traefik_linux-amd64)  with this config:
```toml
defaultEntryPoints = ["http"]

[entryPoints]
  [entryPoints.http]
  address = ":8081"

[file]
  [frontends.frontend-test]
    backend = "backend-test"
  [frontends.frontend-test.routes.test_1]
  rule = "Host:127.0.0.1"
  [backends]
    [backends.backend-test]
        [backends.backend-test.servers.website]
            url = "http://127.0.0.1:8080"
```

And everythings seems to work on `http://127.0.0.1:8081/builds/1`

![capture du 2018-04-24 16-47-13](https://user-images.githubusercontent.com/195661/39195014-3148af20-47df-11e8-8ded-f49b3a7fb383.png)

**Comment by mandrean** [comment]
Hi @Juliens! We talked at KubCon today. So we still seem to have this issue...

*GCP GKE cluster*: `v1.9.6-gke.1` (both master and 3x slaves in node pool)
*helm/tiller*: `v2.9.0` (both)
*traefik*: `v1.5.4`, deployed using `stable/traefik` `v1.29.1` helm chart with Let's Encrypt enabled (dns-01/gcloud)
*concourse*: both `v3.10.0` and `v3.13.0`, deployed using `stable/concourse` `v1.3.2` helm chart with traefik ingress & tls-acme enabled
*GCP GCE node*:
```
 Kernel Version:		4.13.0-1011-gcp
 OS Image:			Ubuntu 16.04.4 LTS
 Operating System:		linux
 Architecture:			amd64
 Container Runtime Version:	docker://17.3.2
 Kubelet Version:		v1.9.6-gke.1
 Kube-Proxy Version:		v1.9.6-gke.1
```

![image](https://user-images.githubusercontent.com/4631744/39549390-f99e5be4-4e5d-11e8-94f8-efda3234a711.png)

**Comment by abellion** [comment]
Same issue here ! Any news ?

**Comment by emcniece** [comment]
Ok folks, I did some heavy debugging today and was unable to reproduce the error.

Example cases: https://gist.github.com/emcniece/11824488f7a4d7ef9e51e4ff1217025a

There are 2 Docker-Compose stacks here: one that uses outdated versions (Concourse 3.9.0, Traefik 1.5.4), and one with updated versions (Concourse 3.14.0, Traefik 1.6.4). Instructions for replicating are included in the [gist README](https://gist.github.com/emcniece/11824488f7a4d7ef9e51e4ff1217025a#file-readme-md). 

Since both stacks do *not* exhibit the error, I won't bother detailing their differences.

This may indicate external interfering services. The gist here runs a local stack, and I can currently only see the issue in my production environment. Over the next week or two I'll be upgrading a number of services, this stack included - will report back once I make that upgrade.

**Comment by starkers** [comment]
I've just hit this..

using both stable helm charts `stable/traefik` and `stable/concourse`

Screenshot:

![image](https://user-images.githubusercontent.com/462087/42130270-aa70ae20-7cd7-11e8-9166-f380d53c7996.png)


## concourse (chart **1.7.0**) installed with:
```
helm upgrade \
  --install \
  --recreate-pods \
  --namespace concourse \
  concourse \
  stable/concourse \
  --set "web.ingress.enabled=true" \
  --set "web.ingress.hosts[0]=concourse.example.com" \
  --set "concourse.externalURL=https://concourse.example.com"
```

## traefik `values.yml` something like this:
Really nothing fancy and I can maybe set this up on a docker-compose for testing if that'll help
it definitely is just things with the stream header thats effected. I just spent best part of a week thinking concourse was crap but it was my beloved traefik all along :-(



```yml
imageTag: v1.6.4-alpine                                                                                                                                                                        
debug:                                                                                                                                                                                         
  enabled: true                                                                                                                                                                                
ssl:                                                                                                                                                                                           
  enabled: true                                                                                                                                                                                
  enforced: true                                                                                                                                                                               
acme:                                                                                                                                                                                          
  enabled: true                                                                                                                                                                                
  email: david@starkers.org                                                                                                                                                                    
  staging: false                                                                                                                                                                               
  persistence.enabled: false                                                                                                                                                                   
  challengeType: http-01                                                                                                                                                                     
rbac:                                                                                                                                                                                          
  enabled: true                                                                                                                                                                                
service:                                                                                                                                                                                       
  annotations:                                                                                                                                                                                 
    external-dns.alpha.kubernetes.io/hostname: '*.example.com.'        
```

Installing helm like this:
```
helm upgrade \
  --namespace traefik \
  --install traefik \
  stable/traefik \
  -f values.yml
```


This effects me on GKE k8s 1.8 and 1.9 both "COS" and "UBUNTU" kernels/hosts.

**Comment by starkers** [comment]
@emcniece I was certain I'd be able to reproduce this easily with docker-compose but then I tried and disturbingly it just always worked.. I even went the route of host networking for traefik to be sure. I can see why you were so frustrated in your tests

I am pretty sure (when I have the time this week) that I can reproduce it by putting https on the frontend of a docker-compose for u.

The first time I encountered this issue was when I hand-rolled a deployment on a single instance and used docker-compose to flesh out what I needed.. naturally I used traefik+acme ssl but I thought it was concourse that was faulty..

It was only later with k8s also that I isolated it the cause to be traefik. I switched ingress controllers and that sorted the issue. (although I'm not happy about a life without traefik)

**TL;DR:** The only common thread that I can think of now is having https on the frontend

**Comment by emcniece** [comment]
@starkers OH, yes - SSL could indeed be the problem! My last test didn't use a certificate as it was all local. I'll try this again on a public instance with https.

**Comment by emcniece** [comment]
@starkers Tested with Traefik+ACME, and it's working fine. Check out https://concourse.concourse.concourse.emc2.build/ if you like - this was created using the [example setup](https://gist.github.com/emcniece/11824488f7a4d7ef9e51e4ff1217025a) with some added Traefik ACME config.

**Comment by starkers** [comment]
this is crazy.. I just reproduced it on a brand new debian:9 box

The exact config files are right here...
-  https://github.com/starkers/snippets/tree/master/traefik-concourse

Its currently running if you'd like to test although I may switch it to http mode to quickly check

**Comment by amnk** [comment]
I have the same with
traefik 1.6.2 with acme
concourse 3.14.1
both installed from stable Helm charts.

**Comment by kamsz** [comment]
Encountered this issue today. Same as in previous posts - stuck on "events" API call.

**Comment by shuraa** [comment]
I've noticed when I've disabled compression on https endpoint, the issue disappeared

```
data:
  traefik.toml: |
    # traefik.toml
    logLevel = "DEBUG"
    defaultEntryPoints = ["http","https"]
    [entryPoints]
      [entryPoints.http]
      address = ":80"
      compress = false
        [entryPoints.http.redirect]
          regex = "^http://(.*)"
          replacement = "https://$1"
      [entryPoints.https]
      address = ":443"
      compress = false
        [entryPoints.https.tls]
          [[entryPoints.https.tls.certificates]]
          CertFile = "/ssl/tls.crt"
          KeyFile = "/ssl/tls.key"
      [entryPoints.traefik]
      address = ":8080"
    [kubernetes]
    [api]
      entryPoint = "traefik"
      dashboard = true
```

**Comment by cuinfo** [comment]
> I've noticed when I've disabled compression on https endpoint, the issue disappeared
> 
> ```
> data:
>   traefik.toml: |
>     # traefik.toml
>     logLevel = "DEBUG"
>     defaultEntryPoints = ["http","https"]
>     [entryPoints]
>       [entryPoints.http]
>       address = ":80"
>       compress = false
>         [entryPoints.http.redirect]
>           regex = "^http://(.*)"
>           replacement = "https://$1"
>       [entryPoints.https]
>       address = ":443"
>       compress = false
>         [entryPoints.https.tls]
>           [[entryPoints.https.tls.certificates]]
>           CertFile = "/ssl/tls.crt"
>           KeyFile = "/ssl/tls.key"
>       [entryPoints.traefik]
>       address = ":8080"
>     [kubernetes]
>     [api]
>       entryPoint = "traefik"
>       dashboard = true
> ```
> **Kubernetes** 1.10.8 (minikube)
> **Traefik** 1.6.6 (helm chart stable/traefik 1.43.0)
> **Concourse** 3.14.1 (helm chart stable/concourse 1.3.1)

it's working

**Comment by Overbryd** [comment]
**Traefik** `1.7.3`

Same problem. When I set `compress = false` the issue disappears.

**Comment by Overbryd** [comment]
Here is a reproducible case: https://gist.github.com/Overbryd/c07aae9f68f487b0bf699c3ec83998b9
Steps to reproduce:

1. Download `docker-compose.yml`
2. Download `config.toml` into `conf/config.toml`
3. `docker-compose up`
4. `curl --compress localhost:8080/events` stalls

Compare against `curl localhost:8080/events` which does not stall (note that compression is off for that request).

**Comment by Ullaakut** [comment]
Hi @Overbryd !

Thank you very much for the stack and steps to reproduce the issue.

I've tried reproducing it using exactly the steps that you provided, and I have been unsuccessful. Both `curl` commands seem to get a response at the same speed.

![ezgif-3-d3d8d936511f](https://user-images.githubusercontent.com/6976628/48201818-907a4600-e363-11e8-8890-3d09723d5657.gif)

I think that I followed the steps correctly, here is my folder's structure:

<img width="180" alt="screenshot 2018-11-08 at 14 38 15" src="https://user-images.githubusercontent.com/6976628/48201982-f6ff6400-e363-11e8-9f25-0a205608d34b.png">

Let me know if I missed something or if there was a mistake in the gist files and I'll try again :)

**Comment by Overbryd** [comment]
@Ullaakut all correct. Mind the trailing slash on your request.
It should be `curl --compressed localhost:8080/events/`.

**Comment by camelpunch** [comment]
Ran into this with Concourse 4.2.1 and Traefik 1.7.4. Took hours to track down.

Steps to reproduce locally:

1. Set a `/etc/hosts` entry for `127.0.0.1 my.url`

2. Run concourse with:
```
sudo $(which concourse_linux_amd64) quickstart \
    --worker-work-dir=/tmp/concourse-workers \
    --postgres-user=concourse \
    --postgres-password=somepassword \
    --main-team-local-user concourse \
    --add-local-user concourse:somepassword \
    --external-url http://my.url \
    --worker-garden-dns-server 8.8.8.8
```

3. Run traefik with `--configfile` of:
```toml
defaultEntryPoints = ["http"]

[entryPoints]

[entryPoints.http]
address = ":80"
compress = true

[file]

[backends]

[backends.concourse]
[backends.concourse.servers]
[backends.concourse.servers.server0]
url = "http://localhost:8080"

[frontends]

[frontends.concourse]
entrypoints = ["http"]
backend = "concourse"
passHostHeader = true

[frontends.concourse.routes]
[frontends.concourse.routes.route0]
rule = "Host:my.url"
```
4. Set up your fly with `fly -t ci login --team-name main --concourse-url http://my.url`. Follow instructions and log in with credentials above e.g. concourse/somepassword

5. Execute a build with `fly -t ci execute -c <(echo '{"platform": linux, "run": {"path": "ls"}}')`

6. Do the same with/without `compress = true`. With compression, the build output is not streamed.

**Comment by demamoh** [comment]
Any update on this? The problem still continues for me on concourse helm chart:  concourse-3.0.1 | traefik-1.56.1.

**Comment by amir20** [comment]
I just ran into this issue. For what its worth, I think the size of the payload makes a difference. When I have less than 100 bytes, it shows no data. But when the payload grows bigger then the data shows up.

**Comment by jPrest** [comment]
> I just ran into this issue. For what its worth, I think the size of the payload makes a difference. When I have less than 100 bytes, it shows no data. But when the payload grows bigger then the data shows up.

I can see the same behaviour on my setup.

**Comment by BenjaminBeck** [comment]
Similar here:
I had a youtrack behind traefik 1.7.
It made a request with "accept: text/event-stream" that stuck on pending state.,.
Setting "compress = false" solved the issue.

**Comment by davinkevin** [comment]
Same problem for me, through traefik integrated inside k3s as ingress controller 😓
